devel/python/python/ert/enkf/key_manager.py

```diff
@@ -100,6 +101,23 @@ def genDataKeys(self):
 
         return self.__gen_data_keys
 
+    def genDataKeysWithObservations(self):
+        """ :rtype: list of str """
+        if self.__gen_data_keys_with_observations is None:
+            enkf_obs = self.ert().getObservations()
+            gen_data_obs_keys = []
+            for obs_vector in enkf_obs:
+                report_step = obs_vector.activeStep()
+                key = obs_vector.getDataKey()
+
+                gen_data_key = "%s@%d" % (key, report_step)
+                if gen_data_key in self.genDataKeys():
```

Comment:
OK - I should have seen this before; but here is a key which does not actually exist in the `ensemble_config` proper - how wise is that?